src/core/data/tensorfield/tensorfield.hpp

```diff
@@ -80,15 +82,18 @@ class TensorField
     //                  start the ResourcesUser interface
     //-------------------------------------------------------------------------
 
-    NO_DISCARD auto getCompileTimeResourcesViewList()
+    void setBuffer(std::nullptr_t ptr)
     {
-        return for_N<N, for_N_R_mode::forward_tuple>(
-            [&](auto i) -> auto& { return components_[i]; });
+        for_N<N, for_N_R_mode::forward_tuple>([&](auto i) { components_[i].setBuffer(nullptr); });
     }
-    NO_DISCARD auto getCompileTimeResourcesViewList() const
+
+    template<typename Fields>
+    void setBuffer(Fields* const fields)
     {
-        return for_N<N, for_N_R_mode::forward_tuple>(
-            [&](auto i) -> auto& { return components_[i]; });
+        if (!fields)
+            throw std::runtime_error("use other fn");
+        for_N<N, for_N_R_mode::forward_tuple>(
```

Comment:
you can delete `, for_N_R_mode::forward_tuple`

there is no return in this *loop*